src/ProximalAlgorithms.jl

```diff
@@ -10,28 +10,29 @@ include("utilities/identity.jl")
 include("utilities/zero.jl")
 include("utilities/conjugate.jl")
 
-abstract type ProximalAlgorithm{I, T} end
+abstract type ProximalAlgorithm{I} end
 
 # The following methods give `ProximalAlgorithm` objects the iterable behavior.
 
-function start(solver::ProximalAlgorithm{I, T})::I where {I, T}
-    initialize(solver)
+function start(solver::ProximalAlgorithm{I})::I where {I}
+    initialize!(solver)
     return zero(I)
 end
 
-function next(solver::ProximalAlgorithm{I, T}, it::I)::Tuple{T, I} where {I, T}
-    point::T = iterate(solver, it)
-    return (point, it + one(I))
+function next(solver::ProximalAlgorithm{I}, it::I)::Tuple{Void, I} where {I}
+    iterate!(solver, it)
+    return (nothing, it + one(I))
```

Comment:
Why this change?

Reply:
I mean, is it related to the crash you observed or the way you fixed it? If there’s no need for it, I would avoid this.

Reply:
it is related: you need a status in the iterator. 

previously it was `point = solver.xbar` for example.  

we could keep `point` like this but then we would need to define an additional function for all solvers that returns `xbar` and initializes `point` before the loop of `run!`.

so I opted for `Void` status instead.

Reply:
But I guess not all algorithms are affected: DouglasRachford and AFBA always perform at least on iterate.

The problem is with forward-backward algorithms, that are implemented in such a way that they compute the initial residual before starting the iterations (in `initialize`). Why not forcing them to always compute at least one iteration instead (by modifying `converged`)?

The nature of the iterator is that of spit out the next element of a certain sequence when asked. It is natural to return in this case the sequence of iterates.

I’m strongly in favor of keeping this return value of `next` (and the type parameters of the basic iterator type) as they were.

Reply:
ok I've changed back the `run!` function in favour of modifying the `converged` function of each solver. 

apparently this was also necessary DouglasRachford and AFBA!